x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/AbstractBlock.java

```
@@ -101,4 +100,54 @@ public BlockFactory blockFactory() {
     public boolean isReleased() {
         return released;
     }
+
+    @Override
+    public final void incRef() {
+        if (references <= 0) {
+            throw new IllegalStateException("can't increase refCount on already released block [" + this + "]");
```

Can we add an assertion `assert false : ...` to make tests fail hard?